### PR TITLE
Added RENDERDOC_PLATFORM_SWITCH define.

### PR DESCRIPTION
**Pull request by kevin-mccullough**
Adding the RENDERDOC_PLATFORM_SWITCH define in the appropriate places.

### CODE REVIEW

File: renderdoc/common/globalconfig.h [hunk 1]
```diff
@@ -104,6 +104,15 @@
...
 #define RDOC_GGP OPTION_ON
 
+#elif defined(RENDERDOC_PLATFORM_SWITCH)
+
+#define RDOC_WIN32 OPTION_OFF
+#define RDOC_ANDROID OPTION_OFF
+#define RDOC_LINUX OPTION_OFF
+#define RDOC_APPLE OPTION_OFF
+#define RDOC_POSIX OPTION_ON
+#define RDOC_GGP OPTION_OFF
```
Comment (baldurk) [inline renderdoc/common/globalconfig.h:114]: We should add a new `RDOC_SWITCH` option here and use that in preference to `RENDERDOC_PLATFORM_SWITCH` wherever possible. It should only be needed either to enable that option here, or in `apidefs.h` which is a public header and so can't include this.

The other uses in common.h, gl_common.h and so on should then use `RDOC_SWITCH`.

File: renderdoc/driver/gl/official/cgl.h [hunk 2]
```diff
@@ -10,40 +10,44 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-  
+
```
Comment (baldurk) [inline renderdoc/driver/gl/official/cgl.h:13]: Looks like this file was clang-format'd by accident?

**Review by baldurk — CHANGES_REQUESTED** [review]
Thanks, I think some of the uses of `RENDERDOC_PLATFORM_SWITCH` can be moved to a less error-prone `RDOC_SWITCH` (since that must be explicitly defined to be enabled or disabled) but otherwise this looks good.

File: renderdoc/driver/gl/official/cgl.h (hunk 2 above)
Comment (kevin-mccullough) [reply renderdoc/driver/gl/official/cgl.h:13]: Sorry, my bad!  I'll fix this with only my changes.

File: renderdoc/common/globalconfig.h (hunk 1 above)
Comment (kevin-mccullough) [reply renderdoc/common/globalconfig.h:114]: I understand.  Will do.

**Comment by kevin-mccullough** [comment]
Closing for the cleaner version: https://github.com/baldurk/renderdoc/pull/1905

**Pull request closed without merging**